build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/InternalDistributionBwcSetupPlugin.java

```diff
@@ -199,6 +226,17 @@ private static List<DistributionProject> resolveArchiveProjects(File checkoutDir
         }).collect(Collectors.toList());
     }
 
+    record StableApiProject(String name, String projectDir) {}
+
+    private static List<StableApiProject> resolveStableProjects() {
+        // TODO: How do we retrieve these programmatically?
```

Comment:
We could avoid having to hard-code the project names _and_ directories. You can resolve the project directory from a `Project`. So something `project.findProject(":libs:elasticsearch-logging").getProjectDir()`. Then we'd just need to hard-code the list of project paths that we care about.

Reply:
Alternatively we could optimistically create these BWC tasks for _all_ projects under `libs`. You can do that via `project.findProject(":libs").getSubProjects()`.

Reply:
I've changed things around so that I'm I'm using a Project object to get names and project directory paths. Is there a way to filter subprojects by a particular property, like `ext.stableApiSince`?

Reply:
I don't think it's critical that we do so. All we're doing  here is registering tasks to build those project. For those that aren't stable plugin projects, we'll simply never use the tasks.